tests/conftest.py

```diff
@@ -57,6 +57,19 @@ def docker_client():
         pytest.skip("docker not available")
 
 
+@pytest.fixture(scope="function")
+def remove_created_images(docker_client):
```

Comment:
I am a little nervous here, how does this ensure that we do not wipe out all the images on the machine? Is that ensured by the `docker_client` fixture?

Even if it supposedly is, I would be much more comfortable if we filtered the image names somehow (e.g. the image name needs to contain `test` or some other unique string.

Reply:
Thinking about this more, this approach is definitely not safe since, since other processes can create docker images in the meantime while the test is running. The logic here would delete these images.

Instead we could perhaps hard-code the name of the test image (`hello-world"), which should always be safe to delete.